lib/messages/abort.ts

```diff
@@ -0,0 +1,55 @@
+import Message from "./message";
+import ValidationSpec from "./validation-spec";
+import {validateDetails, validateMessage, validateReason} from "./util";
+
+interface IAbortFields {
+    readonly details: { [key: string]: any };
+    readonly reason: string;
+}
+
+class AbortFields implements IAbortFields {
+    constructor (private readonly _details: { [key: string]: any }, private readonly _reason: string) {}
+
+    get details(): { [key: string]: any } {
+        return this._details;
+    }
+
+    get reason(): string {
+        return this._reason;
+    }
+}
+
+class Abort implements Message {
+    static TYPE: number = 3;
+    static TEXT: string = "ABORT";
+
+    static VALIDATION_SPEC = new ValidationSpec(
+        3,
+        3,
+        Abort.TEXT,
+        {1: validateDetails, 2: validateReason},
+    )
+
+    constructor(private readonly _fields: IAbortFields) {}
+
+    get details(): { [key: string]: any } {
+        return this._fields.details;
+    }
+
+    get reason(): string {
+        return this._fields.reason;
+    }
+
+    parse(msg: any[]): Abort {
```

Comment:
parse is not a static method. So how do we call it?

Reply:
we should find a way to construct the message object without arguments, then call parse on that object to "fill" those values.

Reply:
In Dart, we have not defined `parse` as part of `Message` class & defined it as static method in messages, should we try a similar approach?